spark/src/test/java/org/apache/iceberg/spark/source/TestFilteredScan.java

```diff
@@ -425,6 +426,21 @@ public void testFilterByNonProjectedColumn() {
     }
   }
 
+  @Test
+  public void testInFilter() {
+    File location = buildPartitionedTable("partitioned_by_data", PARTITION_BY_DATA, "data_ident", "data");
+
+    DataSourceOptions options = new DataSourceOptions(ImmutableMap.of(
+        "path", location.toString())
+    );
+
+    IcebergSource source = new IcebergSource();
+    DataSourceReader reader = source.createReader(options);
+    pushFilters(reader, new In("data", new String[]{"foo", "junction", "brush"}));
```

Comment:
Will it be hard to add tests for a couple of other data types?

Reply:
I agree. Let's add a test for a `DateType` at least to validate that `convertLiteral` is called.

Reply:
👌